pkgs/top-level/all-packages.nix

```
@@ -4855,7 +4855,13 @@ with pkgs;
     enableExtraPlugins = true;
   };
 
-  asciidoctor = callPackage ../tools/typesetting/asciidoctor { };
+  asciidoctor = callPackage ../tools/typesetting/asciidoctor {
+    bundlerApp = bundlerApp.override {
+      # there is no particular reason to pick 3.1 specifically,
+      # but we don't want to be stuck on the currently-default 2.7:
+      ruby = ruby_3_1;
```

```suggestion
      ruby = ruby_3;
```
How about we add an alias for 3.X to not accidentally forget this for 3.2?

---

`ruby_3` does not exist yet, and if we want to add it I think we should add it in a separate PR. It is not completely obvious to me that it is desirable, as it introduces the same problem we have with `ruby` right now, i.e. that we don't dare to update it to the newer ruby because it would cause many rebuilds and it's not reasonably easy to test all impacted packages. 

Perhaps explicitly referencing `ruby_3_1` is actually better, perhaps not, but in any case introducing `ruby_3` does not belong in *this* PR IMHO.

---

> if we want to add it I think we should add it in a separate PR

Thats not really a problem, we can do it in this one with no problems. 

> that we don't dare to update it to the newer ruby because it would cause many rebuilds and it's not reasonably easy to test all impacted packages.

The rebuild count is fine, we just target staging. Also Jumping from 2.X to 3.X is way more impactful than from 3.1 to 3.2 and if a small amount of packages breaks that is okay. We cna fix them in staging-next.

---

> > if we want to add it I think we should add it in a separate PR
> 
> Thats not really a problem, we can do it in this one with no problems.

I know I can, but I didn't want to, since I think this is a decision that the ruby ecosytem maintainers (like @marsam) should be involved in - I didn't want to make that change without their input.

---

Since there were no objections from the ruby maintainers pinged here, I went ahead and made the change suggested by @SuperSandro2000 - though TBH I still prefer to keep PR's smaller in scope, if not for technical reasons then for social ones.

---

I think adding ruby_3 may be less useful than its Python counterpart, since ruby 3 didn't represent a huge breakage and its adoption is doing well; also ruby 2.7 is going EOL on March 2023, so we probably are going remove it for 23.05.

---

> TBH I still prefer to keep PR's smaller in scope, if not for technical reasons

which means more PRs and more notifications.